src/bolos/cx_mpi.c

```diff
@@ -984,12 +984,23 @@ cx_err_t cx_mpi_rand(cx_mpi_t *x)
 cx_err_t cx_mpi_rng(cx_mpi_t *r, const cx_mpi_t *n)
 {
   cx_err_t error;
+  cx_mpi_t *tmp;
```

Comment:
Instead of introducing a temporary BN, it might be easier to call `BN_rand_range` in a loop until the result is different than zero.

Reply:
That's smart and avoid loosing _resolution_ in the random range: I like it; thanks for the suggestion ;-)